### ISSUE TITLE
'AddODataQueryFilter' extension should also consider 'IEnumerable' return types

### ISSUE THREAD

**Issue by julealgon** [OData/AspNetCoreOData#16]
The `ODataServiceCollectionExtensions.AddODataQueryFilter` extension method currently adds the `EnableQuery` logic to all methods that expose an `IQueryable` or `IQueryable<T>` return type automatically. However, methods returning `IEnumerable` and `IEnumerable<T>` also support the `[EnableQuery]` attribute today. Considering that, I feel the current implementation of this extension is a bit confusing and inconsistent.

My suggestion would be to also consider the `IEnumerable` return types in this extension.

I'm currently using `Microsoft.AspNetCore.OData` Version=7.5.0.0

**Comment by xuzhg** [comment]
AddODataQueryFilter is not at 8.0 preview yet.

**Comment by julealgon** [comment]
> AddODataQueryFilter is not at 8.0 preview yet.

I'm not sure I follow your comment @xuzhg . Can you elaborate? As I mentioned above, I'm using version 7.5.0.0 of the library. Are you implying that this method has been removed from the API?

**Comment by xuzhg** [comment]
@julealgon Yes. AddODataQueryFilter  is not in 8.0 preview. Let me transfer this issue to "webapi" repo.
